.github/workflows/soak.yml

```diff
@@ -0,0 +1,244 @@
+    needs: [build-baseline-image]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download baseline-soak artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: baseline-soak
+
+      - name: Run baseline experiment
+        run: |
+          mkdir -p /tmp/${{ matrix.target }}/
+          ./soaks/bin/soak_one.sh "false" ${{ matrix.target }} "baseline" `cat baseline-soak/sha` /tmp/${{ matrix.target }}
+
+      - name: Upload timing captures
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.target}}-captures
+          path: /tmp/${{ matrix.target }}
+
+  soak-comparison:
+    name: Soak (${{ matrix.target }}) - comparison
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak]
+    needs: [build-comparison-image]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download baseline-soak artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: baseline-soak
+
+      - name: Run comparison experiment
+        run: |
+          mkdir -p /tmp/${{ matrix.target }}/
+          ./soaks/bin/soak_one.sh "false" ${{ matrix.target }} "comparison" ${{ github.sha }} /tmp/${{ matrix.target }}
+
+      - name: Upload timing captures
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.target}}-captures
+          path: /tmp/${{ matrix.target }}
+
+  analyze-results:
+    name: Soak analysis (${{ matrix.target }})
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak] # could be general if we move away from miller
+    needs: [soak-baseline, soak-comparison]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download captures artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: ${{ matrix.target }}-captures
+
+      - name: Analyze captures
+        run: |
+          ./soaks/bin/analyze_experiment.sh ${{ matrix.target }}-captures
+
+  observer:
```

Comment:
I think this job is missing `needs`

Reply:
I don't believe so? The `observer` container doesn't depend on the other jobs in this workflow and is not changed at runtime in the soak terraform.